nvme_gw_cli.py

```diff
@@ -211,20 +216,74 @@ def create_namespace(self, args):
         except Exception as error:
             self.logger.error(f"Failed to add namespace: \n {error}")
 
-    @cli.cmd([argument("-n", "--subnqn", help="Subsystem NQN", required=True)])
-    def allow_any_hosts(self, args):
-        """Allows any host to access a subsystem."""
+    @cli.cmd([
+        argument("-n", "--subnqn", help="Subsystem NQN", required=True),
+        argument("-i", "--nsid", help="Namespace ID", type=int, required=True),
+    ])
+    def delete_namespace(self, args):
+        """Deletes a namespace from a previously created subsystem."""
+
+        try:
+            delete_req = pb2.ns_delete_req(subsystem_nqn=args.subnqn,
+                                           nsid=args.nsid)
+            ret = self.stub.nvmf_subsystem_remove_ns(delete_req)
+            self.logger.info(f"Deleted namespace {delete_req.nsid}: {ret}")
+        except Exception as error:
+            self.logger.error(f"Failed to remove namespace: \n {error}")
+
+    @cli.cmd([
+        argument("-n", "--subnqn", help="Subsystem NQN", required=True),
+        argument("-t", "--host", help="Host NQN", required=True),
+    ])
+    def add_host(self, args):
+        """Adds a host to a previously created subsystem."""
+
+        if args.host == "*":
+            allow_any_host = True
+
+        try:
+            create_req = pb2.subsystem_add_host_req(
+                subsystem_nqn=args.subnqn,
+                host_nqn=args.host,
+                allow_any_host=allow_any_host)
```

Comment:
This references an undefined variable when `--host` is not `*`.  But rather than making sure that `allow_any_host` is always defined here, I'm wondering if it is needed at all.  You got rid of `--any` option on the CLI, why not get rid of `allow_any_host` field in the protocol as well and change the server to discriminate between `nvmf_subsystem_allow_any_host` and `nvmf_subsystem_add_host` based on `request.host_nqn == "*"`?

Reply:
Also, this means that adding individual hosts isn't being tested.  Please add that scenario to the test suite and let's prioritize committing it to the repo so that it can at least be run manually (and later in Jenkins).